src/parser.rs

```diff
@@ -205,6 +211,11 @@ pub fn parse_lcov(
                     });
 
                 if key.is_none() {
+                    if ignore_parsing_error {
```

Comment:
nit: In order to have too much redundant code, I'd use a macro, in order to slightly simplify the code.